storage/core/src/adaptor/locking/mod.rs

```diff
@@ -36,7 +38,15 @@ impl<'tx, T: ReadOps> ReadOps for TxRo<'tx, T> {
     }
 }
 
-impl<'tx, T: ReadOps> traits::TxRo for TxRo<'tx, T> {}
+impl<'tx, 'i, T: PrefixIter<'i>> PrefixIter<'i> for TxRo<'tx, T> {
+    type Iterator = T::Iterator;
+
+    fn prefix_iter<'m: 'i>(&'m self, idx: DbIndex, prefix: Data) -> crate::Result<Self::Iterator> {
+        self.0.prefix_iter(idx, prefix)
+    }
+}
+
+impl<'tx, T: ReadOps> backend::TxRo for TxRo<'tx, T> {}
 
 // Tracker for database changes
 type DeltaMap = BTreeMap<Data, Option<Data>>;
```

Comment:
I think the correct way to do this is by having an Operation enum, just like in TransactionVerifier. The bonus you get there is that you can support caching read-operations by adding an extra arm, which you can exclude from writing back to db when you commit.

Reply:
Can have a look into it but preferably not as a part of the present PR. There will have to be some thought around how to mutate the data structure.The `get()` method now takes `&self`, not  `&mut self` so some decision whether to introduce mutability into the interface or use some interior mutability mechanism would have to be made.

Reply:
That's fair. I don't mind doing this in another PR. But I also want to point out that even without reads, mapping Some to Write and None to Erase is confusing. Let's fix this some times later with or without reads.